packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkNativeInitialization.kt

```diff
@@ -0,0 +1,341 @@
+        return when (siteLower) {
+            "us1", "us" -> DatadogSite.US1
+            "eu1", "eu" -> DatadogSite.EU1
+            "us3" -> DatadogSite.US3
+            "us5" -> DatadogSite.US5
+            "us1_fed", "gov" -> DatadogSite.US1_FED
+            "ap1" -> DatadogSite.AP1
+            else -> DatadogSite.US1
+        }
+    }
+
+    private fun buildVitalUpdateFrequency(vitalsUpdateFrequency: String?): VitalsUpdateFrequency {
+        val vitalUpdateFrequencyLower = vitalsUpdateFrequency?.lowercase(Locale.US)
+        return when (vitalUpdateFrequencyLower) {
+            "never" -> VitalsUpdateFrequency.NEVER
+            "rare" -> VitalsUpdateFrequency.RARE
+            "average" -> VitalsUpdateFrequency.AVERAGE
+            "frequent" -> VitalsUpdateFrequency.FREQUENT
+            else -> VitalsUpdateFrequency.AVERAGE
+        }
+    }
+
+    private fun buildUploadFrequency(uploadFrequency: String?): UploadFrequency {
+        return when (uploadFrequency?.lowercase(Locale.US)) {
+            "rare" -> UploadFrequency.RARE
+            "average" -> UploadFrequency.AVERAGE
+            "frequent" -> UploadFrequency.FREQUENT
+            else -> UploadFrequency.AVERAGE
+        }
+    }
+
+    private fun buildBatchSize(batchSize: String?): BatchSize {
+        return when (batchSize?.lowercase(Locale.US)) {
+            "small" -> BatchSize.SMALL
+            "medium" -> BatchSize.MEDIUM
+            "large" -> BatchSize.LARGE
+            else -> BatchSize.MEDIUM
+        }
+    }
+
+    internal fun getConfigurationFromJSONFile(): DdSdkConfiguration {
+        try {
+            val jsonString = jsonFileReader.parseAssetsJSONFile(appContext, "datadog-configuration.json")
+
+            val configuration = Gson().fromJson(jsonString, JSONConfigurationFile::class.java).configuration
+
+            return configuration.asDdSdkConfiguration()
+        } catch (@Suppress("TooGenericExceptionCaught") exception: Exception) {
+            throw BadConfigurationException(exception)
+        }
+    }
+
+    companion object {
+        /**
+         * Initializes the Datadog React Native SDK from your MainApplication.
+         *
+         * @param appContext: The application context of your React Native application.
+         */
+        @JvmStatic
+        fun initFromNative(appContext: Context) {
+            val nativeInitialization = DdSdkNativeInitialization(appContext.applicationContext)
+            try {
+                nativeInitialization.initialize(nativeInitialization.getConfigurationFromJSONFile())
+            } catch (@Suppress("TooGenericExceptionCaught") error: Exception) {
+                Log.w(
+                    DdSdkNativeInitialization::class.java.canonicalName,
+                    "Failed to initialize the Datadog SDK: $error"
+                )
+            }
+        }
+    }
+}
+
+internal class BadConfigurationException(exception: java.lang.Exception) : Exception(exception.message)
```

Comment:
it is a runtime exception, so it should extend `RuntimeException`. Also we need to keep the original exception stacktrace, so it is better to pass it instead of `exception.message`:

```suggestion
internal class BadConfigurationException(exception: Exception) : RuntimeException(exception)

```